wikimedia/wikipedia-ios
# Disable full screen presentation of long-pressed timeline images

https://phabricator.wikimedia.org/T252922

## Existing Behavior

I unexpectedly had to do a deep dive to get a better idea of what's up here, but I could still be missing something completely obvious. From my research and experimentation so far, here's my current understanding of what was happening. With "timeline" images in an article (that aren't "links" or included in the page's [media list](https://en.wikipedia.org/api/rest_v1/page/media-list/)), long-pressing them triggers WebKit's default link previewing/context menu instead of our custom one – it doesn't trigger [this delegate method](https://github.com/wikimedia/wikipedia-ios/blob/83961737b9868613e1ffea6366d63510bb082dd2/Wikipedia/Code/ArticleViewController%2BLinkPreviewing.swift#L197). It seems we aren't alone in experiencing this custom context menu issue. Tapping the timeline image preview in this default context menu results in the user being stuck in a non-scrollable or dismissible full screen view.

## Testing

Here are a few articles you can reproduce the issue with: 
https://en.wikipedia.org/wiki/Terrace_House:_Opening_New_Doors
https://en.wikipedia.org/wiki/1918_Atlantic_hurricane_season
https://en.wikipedia.org/wiki/Timeline_of_computing

1. Open any article with a graphical timeline
2. Long-press the timeline image
3. Tap the preview of the image in the context menu
4. Observe it now dismisses the context menu preview instead of pushing into a non-dismissible view controller like before

## Question

It's currently WIP because I'm not sure how to handle the share case. When trying to share a timeline image (long-press a timeline image to show context menu, tap Share), it's trying to share our scheme handled URL (`app://`) instead of either the resolved resource URL or just resource itself. This isn't a 6.6 regression - it occurred with 6.5.1 as well.

Maybe an easy solve is if `mobile-html` just prepended the `https` protocol for timeline URLs? I'm unclear what would be best – any ideas @tonisevener?
